### DISCUSSION TITLE
Customizing Operation/Span for JAX-RS

### DISCUSSION THREAD

**Discussion by dmossakowski** [signalfx/splunk-otel-java#578] [General]
Hi, in our application we use very few entry points which are like:

```
        @POST
	@Path("/{method: .+}")
	public Object postData
```

When this is mapped to /ws we see a single operation as  ws/{method: .+} and all requests against it.

As you can imagine this is not very useful. I know that this could be done in code to set name of the Span but is there a way to configure this to use http.target or http.url information?

**Comment by iNikem** [comment]
This follows [Otel semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions) which says that span name for HTTP server spans should be the route that the request was mapped to. Currently there is no built-in way to change this.

**Comment by dmossakowski** [comment]
Thanks Nikita, I saw somewhere that there was a way to configure routes. Is that a possible avenue for doing this? Could we manually specify routes based on a dynamic condition where http.target=ws/someMethod  ?
someMethod.route
  http.target=ws/someMethod
someMethod2.route
  http.target=ws/someMethod2

**Reply by iNikem** [reply]
There is no such functionality atm.

**Reply by dmossakowski** [reply]
Thanks Nikita, what would be your recommendation for us increase the cardinality of operations? Can we customize the operation in code itself? How would we do that? Span.current().updateName() or something?

https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/Span.java

**Reply by iNikem** [reply]
Yes, `Span.current().updateName()` is totally legal way to do that. Please try, if this works for you.
